app/api/api_v1/endpoints/assignment_router.py

```diff
@@ -41,4 +42,22 @@ def get_assignment_submissions(
         .filter_by(assignment_id=assignment_id) \
         .join(StudentModel) \
         .filter(StudentModel.student_onyen == onyen) \
-        .all()
\ No newline at end of file
+        .all()
+
+@router.post("/assignment/{assignment_id}/deadline", response_model=AssignmentSchema)
+def set_assignment_deadline(
+    *,
+    db: Session = Depends(get_db),
+    assignment_id: int,
+    new_due_date: str
+):
+    assignment = db.query(AssignmentModel).filter_by(id=assignment_id).first()
+    if assignment is None:
+        raise HTTPException(status_code=404, detail="Assignment does not exist")
+    
+    # convert the input param string to a datetime object
+    new_due_date = datetime.strptime(new_due_date, '%Y-%m-%d %H:%M:%S')
```

Comment:
Gotta catch the `ValueError` that's thrown if the new_due_date or format can’t be parsed. If it's thrown I would raise a new HTTPException with a 400 error code (bad request).

Reply:
I would also move this to before the db query, so we can tell the user they have the wrong input format before we even query the db.

Reply:
yeah good call on validating it earlier, before the query

Reply:
You should be able to use `datetime.datetime` directly in the arguments to accept a timezoned datetime, and FastAPI will handle the validation and parsing.

Reply:
using datetime now, and the validation should be handled within the json body